src/liteclient.cc

```diff
@@ -105,10 +94,28 @@ LiteClient::LiteClient(Config& config_in)
     throw std::runtime_error("Unsupported package manager type: " + config.pacman.type);
   }
 
-  writeCurrentTarget(current_target);
-  if (target_installation_result.result_code != data::ResultCode::Numeric::kAlreadyProcessed) {
-    notifyInstallFinished(target_been_applied, target_installation_result);
+  {
+    // finalize a pending update if any
+    boost::optional<Uptane::Target> pending_target;
+    storage->loadInstalledVersions("", nullptr, &pending_target);
+
+    if (!!pending_target) {
+      data::InstallationResult update_finalization_result = package_manager_->finalizeInstall(*pending_target);
+      if (update_finalization_result.isSuccess()) {
+        LOG_INFO << "Marking target install complete for: " << *pending_target;
+        storage->saveInstalledVersion("", *pending_target, InstalledVersionUpdateMode::kCurrent);
+      }
+
+      if (update_finalization_result.result_code != data::ResultCode::Numeric::kAlreadyProcessed ||
+          update_finalization_result.result_code != data::ResultCode::Numeric::kNeedCompletion) {
+        notifyInstallFinished(*pending_target, update_finalization_result);
```

Comment:
this will hit our backend without any `x-ats-target: unknown`?

Reply:
Not sure if I fully understand the comment, but just in case there is the setting of x-ats-target to "unknown" at line #70.
Then this header is updated whenever an installation was successful, either in `daemon_main` or in LiteClient ctor after a reboot and successful finalization.

Reply:
i guess my question should be: does sending an installFinished event with an x-ats-target=unknown header cause any unintended problems on our backend. Is the device.status correctly set, etc?

Reply:
Right, I remember that the reason to call update_reguest_headers before was to make sure x-ats-target had at least the current installed version string.

Reply:
Yeah, I got it, I just thought about something else here...
It looks like `headers.emplace_back("x-ats-target: unknown");` at line #70 is redundant/useless.